### PR TITLE
feat(sso): classify an SSO configuration as authentication only

### PR DESCRIPTION
**Pull request by dorsha**
## Description

Adds `authenticationOnly` to the SSO settings objects — `SSOSAMLSettings`, `SSOSAMLSettingsByMetadata` and `SSOOIDCSettings` — and surfaces it on `SSOTenantSettingsResponse`.

A connection classified this way verifies a person's identity without creating, updating or signing in a user: the login authenticates against the IdP and returns the IdP response, with no user and no session behind it. See descope/backend#2713.

It is a `*bool`. Left nil it is not sent at all, so an ordinary settings save keeps whatever the tenant configured instead of clearing it; `false` clears it. Setting it through any one protocol classifies the whole connection, including the tenant's default one.

## Tests

`ConfigureSAMLSettings` and `ConfigureOIDCSettings` send it when set, send `false` explicitly so the classification can be cleared rather than dropped as a falsy value, and omit it entirely when the caller says nothing.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: descope/internal/mgmt/sso.go [hunk 1]
```diff
@@ -109,6 +109,11 @@ func (s *sso) ConfigureSAMLSettings(ctx context.Context, tenantID string, settin
 	if settings.ConfigFGATenantIDResourceSuffix != "" {
 		req["settings"].(map[string]any)["configFGATenantIDResourceSuffix"] = settings.ConfigFGATenantIDResourceSuffix
 	}
+	if settings.AuthenticationOnly != nil {
+		// Sent only when set: the server keeps the stored classification for a request that says
+		// nothing, which is what stops an ordinary settings save from clearing it.
+		req["settings"].(map[string]any)["authenticationOnly"] = *settings.AuthenticationOnly
```
[the remark this answers is not in the document]
Comment (dorsha) [reply descope/internal/mgmt/sso.go]: Both do hold, for different fields, and the comment did not say so. `disableSignRequest` is a plain proto bool, so the server cannot tell false from unset and the SDK has to send it every time. `authenticationOnly` is `optional` on the wire, so an absent field means "keep what is stored" - a deliberate field-level exception to the full-replacement rule, because a settings edit from a client that predates the field would otherwise put a verification connection back in the business of creating users.

Documented in dc84769, at both call sites and on all three settings structs. The backend side is covered by a test asserting a save that omits the field leaves the stored classification alone, and by the same case end to end in descope/integrationtests#14895.

File: descope/types.go [hunk 2]
```diff
@@ -232,13 +248,24 @@ type SSOOIDCSettings struct {
...
 	FgaMappings          map[string]*FGAGroupMapping `json:"fgaMappings,omitempty"`
 	LastSuccessTestTime  int32                       `json:"lastSuccessTestTime,omitempty"` // epoch seconds of the last successful SSO test login on this configuration (read-only, ignored on configure)
+
+	// AuthenticationOnly classifies the configuration as verifying identity only: a login through it
+	// does not create, update or sign in a user, and returns the IdP response instead of a session.
+	//
+	// It is a pointer because, unlike the rest of this object, the server treats it as optional rather
+	// than as part of the full replacement: nil is not sent and leaves whatever is stored, so an
+	// ordinary settings save cannot clear a classification by omission. False clears it.
+	AuthenticationOnly *bool `json:"authenticationOnly,omitempty"`
```
Comment (dorsha) [inline descope/types.go]: On load responses this same struct is the `Oidc` field and the backend never sets it there; only the top-level `SSOTenantSettingsResponse.AuthenticationOnly` is populated. One line saying "on load read SSOTenantSettingsResponse.AuthenticationOnly" avoids a silent false.

**Review by dorsha — COMMENTED** [review]
Review against backend#2713. The hand-built request maps send the field only when non-nil and deref to an explicit false, `ConfigureOIDCSettings` passes the struct so omitempty gives the same behavior, and both `LoadSettings` and `LoadAllSettings` decode into `SSOTenantSettingsResponse` on the v2 routes. Nothing blocking.

- `ConfigureXAASettings` (descope/internal/mgmt/sso.go:457-466) builds its own map and `SSOXAASettings` has no field, while the backend accepts `authenticationOnly` on `ConfigureXAASettingsRequest`. Not a correctness bug (a SAML or OIDC save stamps the XAA row too) but a surface gap; node-sdk#807 has the same one.
- README documents `DisableSignRequest` in the SSO section but not this; add it with the pointer semantics (nil keeps stored, `&false` clears).
- Tests: no case for `ConfigureSAMLSettingsByMetadata` sending it (its own hand map, the likeliest to regress), and none that `LoadSettings` / `LoadAllSettings` decode it.

File: descope/types.go (hunk 2 above)
Comment (dorsha) [reply descope/types.go]: Added in 772ced8, on both ends: the nested field says write-only and points at the top-level one, and the top-level one says it is the one to read. The backend proto carries the same note now.

**Comment by dorsha** [comment]
Added `AuthenticationOnly` to `SSOXAASettings` and wired it into `ConfigureXAASettings` — that endpoint builds its request map by hand, so without this there was no way to classify a configuration through Cross-App Access, and the three protocols disagreed on what the SDK could express.

Also added the tests that were missing: the by-metadata save sends it and omits it when nil, the Cross-App Access save does the same, and `LoadSettings`/`LoadAllSettings` decode `authenticationOnly` off the top-level response field rather than the write-only one nested under `Oidc`. README documents the pointer semantics (nil keeps stored, `&false` clears, any one protocol classifies the whole configuration).
